adapter/i_o.py

```diff
@@ -854,3 +856,44 @@ def process_column_labels(self, list_of_labels):
         ]
 
         return list_of_cleaned_labels
+
+
+def to_pickle(obj, out_path):
+    """Pickle a Python object and save it to a gzip-compressed file at the specified output path.
+
+    Args:
+        obj: The Python object or objects to pickle.
+        out_path: The output path for the pickled object(s).
+
+    Raises:
+        IOError: If there is an error writing to the output file.
+
+    """
+    with gzip.open(out_path, "wb", compresslevel=9) as f:
```

Comment:
Was 9 used here it gives the maximum compression? Compression level 9 also gives the slowest speed though. I'd recommend the default compression level , which is a good compromise between speed and compression ratio.